### PR TITLE
fix(ui-components): adds skip_load_on_open field and adds migration

### PR DESCRIPTION
**Pull request by scefali**
It turns out that the `async` field we have for UI components does not behave in the way that the docs explain. It turns out the only thing it actually does is skip loading the options when the page first loads (instead, it waits until the user starts typing). This PR fixes this issue by doing the following:
1. Implements the `async` field properly in that it controls whether we fetch more options when the user types. In order to maintain legacy behavior, the default value is `true`.
2. Creating a new option `skip_load_on_open` which acts the same was as `async` did before.
3. Migrate all instances of `async` to `skip_load_on_open` in order to maintain the current form behavior for existing applications.

As far as the migration goes, the `SentryAppComponent` table is pretty small.

Docs PR to go with this: https://github.com/getsentry/sentry-docs/pull/1930

### CODE REVIEW

File: src/sentry/api/validators/sentry_apps/schema.py [hunk 1]
```diff
@@ -43,6 +43,8 @@
                 "type": {"type": "string", "enum": ["select"]},
                 "label": {"type": "string"},
                 "name": {"type": "string"},
+                "async": {"type": "boolean"},
```
Comment (MeredithAnya) [inline src/sentry/api/validators/sentry_apps/schema.py:46]: do you think we could add these options in the `test_issue_link` test?

Comment (scefali) [reply src/sentry/api/validators/sentry_apps/schema.py:46]: @MeredithAnya Which test is this? can't find it.

Comment (MeredithAnya) [reply src/sentry/api/validators/sentry_apps/schema.py:46]: `validators/sentry_apps/test_issue_link.py` p sure

Comment (scefali) [reply src/sentry/api/validators/sentry_apps/schema.py:46]: @MeredithAnya My bad, I didn't realize it was a file. Yes, I will update it.

**Comment by MeredithAnya** [comment]
If someone from @getsentry/owners-migrations could give this some 👀  that would be good, I think overall this looks good to me tho

**Review by wedamija — COMMENTED** [review]
I don't know the logic here so can't comment on that, but the migration looks safe, only 64 rows and so should be fine to run on deploy.

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**